src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -309,6 +320,19 @@ export class KycService {
     return KycStepMapper.toKycResult(kycStep);
   }
 
+  async updateSignatoryPowerData(kycHash: string, stepId: number, data: KycSignatoryPowerData): Promise<KycResultDto> {
+    let user = await this.getUser(kycHash);
+    const kycStep = user.getPendingStepOrThrow(stepId);
+
+    user = await this.userDataService.updateUserDataInternal(user, data);
+
+    user = user.internalReviewStep(kycStep);
+    await this.createStepLog(user, kycStep);
+    await this.updateProgress(user, false);
+
+    return KycStepMapper.toKycResult(kycStep);
+  }
+
```

Comment:
Merge `updateSignatoryPowerData`, `updateNationalityData` and `updateLegalEntityData` together in one method. It's the same logic for all 3 methods, or?

Reply:
I agree, makes sense 👍

Reply:
Would it make sense to do the same inside the `kyc.controller` or is it cleaner to keep things separate there?

Reply:
Yeah thought about that too. I'm not sure if it makes more sense like that in kyc controller for `outside` devs to understand 🤔 maybe @davidleomay has an opinion on that. Therefore in the controller I would be fine with both ways ✌️

Reply:
I would keep it in the controller 🙂